Update test262

Constellation opened the pull request:
#### f2f6b5d76ff5107378c7fd186bf35ba981c4136b
<pre>
Update test262
<a href="https://bugs.webkit.org/show_bug.cgi?id=248010">https://bugs.webkit.org/show_bug.cgi?id=248010</a>
rdar://102446984

Reviewed by Ross Kirsling and Mark Lam.

* JSTests/test262/expectations.yaml:

Canonical link: <a href="https://commits.webkit.org/256770@main">https://commits.webkit.org/256770@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ee207af3732043eacd8f145c92c4c04dc284a591

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96805 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6072 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29899 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106331 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6285 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34801 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89190 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103033 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102480 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/4707 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83418 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/31663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/86540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/74616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/87748 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/105 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19897 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/83191 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/96 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/21315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/28373 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/4887 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/85876 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2259 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1323 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/40611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/45/builds/19351 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

MenloDorian reviewed (approved):
rs=me

rkirsling commented on JSTests/test262/expectations.yaml:
```diff
@@ -5,6 +5,9 @@ test/annexB/built-ins/Function/createdynfn-html-close-comment-params.js:
 test/annexB/built-ins/Function/createdynfn-html-open-comment-params.js:
   default: "SyntaxError: Unexpected token '}'. Expected a parameter pattern or a ')' in parameter list."
   strict mode: "SyntaxError: Unexpected token '}'. Expected a parameter pattern or a ')' in parameter list."
+test/annexB/built-ins/RegExp/prototype/compile/duplicate-named-capturing-groups-syntax.js:
```
I think there's a `regexp-duplicate-named-groups` feature for this.

Constellation replied:
Added.

rkirsling commented on JSTests/test262/expectations.yaml:
```diff
@@ -1353,12 +1377,210 @@ test/built-ins/RegExp/prototype/Symbol.search/u-lastindex-advance.js:
+test/built-ins/Temporal/Duration/compare/order-of-operations.js:
+  default: 'Test262Error: Expected [get one.days, get one.days.valueOf, call one.days.valueOf, get one.hours, get one.hours.valueOf, call one.hours.valueOf, get one.microseconds, get one.microseconds.valueOf, call one.microseconds.valueOf, get one.milliseconds, get one.milliseconds.valueOf, call one.milliseconds.valueOf, get one.minutes, get one.minutes.valueOf, call one.minutes.valueOf, get one. ...
+  strict mode: 'Test262Error: Expected [get one.days, get one.days.valueOf, call one.days.valueOf, get one.hours, get one.hours.valueOf, call one.hours.valueOf, get one.microseconds, get one.microseconds.valueOf, call one.microseconds.valueOf, get one.milliseconds, get one.milliseconds.valueOf, call one.milliseconds.valueOf, get one.minutes, get one.minutes.valueOf, call one.minutes.valueOf, get  ...
+test/built-ins/Temporal/Duration/compare/precision-exact-mathematical-values-1.js:
```
We'll need to sort through these, though I can do it in a follow-up if you prefer.

rkirsling reviewed (approved):
r=me with comments

Constellation committed:
Update test262
https://bugs.webkit.org/show_bug.cgi?id=248010
rdar://102446984

Reviewed by Ross Kirsling and Mark Lam.

* JSTests/test262/expectations.yaml:

Canonical link: https://commits.webkit.org/256770@main
